python/dolfinx/wrappers/nls.cpp

```diff
@@ -34,6 +34,9 @@ void nls(py::module& m)
       .def("solve", &dolfinx::nls::NewtonSolver::solve)
       .def_readwrite("atol", &dolfinx::nls::NewtonSolver::atol)
       .def_readwrite("rtol", &dolfinx::nls::NewtonSolver::rtol)
+      .def_readwrite("error_on_nonconvergence", 
+                     &dolfinx::nls::NewtonSolver::error_on_nonconvergence)
+      .def_readwrite("record", &dolfinx::nls::NewtonSolver::record)
```

Comment:
```suggestion
      .def_readwrite("report", &dolfinx::nls::NewtonSolver::report)
```
Please also merge main into your branch and resolve possible merge conflicts.